nautobot/extras/forms/forms.py

```diff
@@ -1013,6 +1013,10 @@ class JobHookForm(BootstrapMixin, forms.ModelForm):
     content_types = MultipleContentTypeField(
         queryset=ChangeLoggedModelsQuery().as_queryset(), required=True, label="Content Type(s)"
     )
+    job = DynamicModelChoiceField(
+        queryset=Job.objects.filter(is_job_hook_receiver=True),
```

Comment:
This is the only case I have seen that we needed `.filter` on the `queryset=` argument. I think the `query_params` is sufficient to filter the choices for this field. Do double check me